apps/src/studio/collidable.js

```diff
@@ -104,6 +108,157 @@ Collidable.prototype.bounce = function () {
   }
 };
 
+function getRandomInt(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+Collidable.prototype.roamGrid = function(type) {
+
+  // Do we have an active location in grid coords?  If not, determine it.
+  if (this.gridX === undefined) {
+    this.gridX = Math.floor(this.x / Studio.SQUARE_SIZE);
+    this.gridY = Math.floor(this.y / Studio.SQUARE_SIZE);
+  }
```

Comment:
Initializing all fields in the constructor is also more likely to trigger "hidden classes" optimizations in the JavaScript runtime!